site/content/entry/yuzu-progress-report-aug-2023/index.md

```diff
@@ -0,0 +1,529 @@
+This avoids reading the buffer back from memory, which can be expensive.
+So Vulkan does not have a direct equivalent of masked clears, and requires clearing the entire "aspect" of an image (colour, depth, or stencil) at once.
+
+`Red Dead Redemption` was using a masked clear on its stencil buffer. This is why the bug was only affecting grass and bushes: the game was trying to keep those values intact with the mask.
+But since we translated clears directly from the game to Vulkan clears, we were just wiping out the entire stencil buffer, breaking the bush and grass rendering in the process.
+
+The solution we came up with for Vulkan is similar to a trick [we previously used for Persona 5 Royal](https://github.com/yuzu-emu/yuzu/pull/9631): use a regular draw, rather than a clear.
+With a regular draw, we are able to use the built-in hardware support for applying a stencil mask, and produce the same result as OpenGL.
+
+
+The new system in place works by drawing a fullscreen triangle, which allows the regular stencil hardware to handle masking and buffer writing. Thanks to this, {{< gh-hovercard "11320" "rendering issues in grass and bushes in Red Dead Redemption are now solved." >}}
+Howdy! You get correct rendering directly to your face, like it or not.
+
+{{< single-title-imgs
+    "You have some tequilas, then Juan decides he is an adventurer (Red Dead Redemption)"
+    "./rdr.png"
+    "./juan.png"
+    >}}
+
+Blinkhawk and Maide also spotted an old regression in how indirect compute dispatch is emulated.
+For the uninitiated, indirect dispatch is a technique used to allow the GPU to get the amount of work it needs to perform from a result generated by another compute shader or GPU operation, instead of being passed by the CPU at the moment of recording.
+Such an independent and strong Tegra X1 GPU. Look at it grow.
+
+Thanks to this {{< gh-hovercard "11383" "late intervention," >}} `Mario + Rabbids Kingdom Battle`, `Mario + Rabbids Sparks of Hope`, and `Sea of Solitude` now render much better: shadows in the Rabbids series have the proper level of shade, and environments in Sea of Solitude are now visible.
+
+{{< single-title-imgs
+    "This one took a while, eh (Mario + Rabbids Kingdom Battle & Sea of Solitude)"
+    "./mr.png"
+    "./sos.png"
+    >}}
+
+The water in Sea of Solitude requires emulating other very tricky aspects of the Tegra X1, so that’s homework for later.
+
+As Maide also has learnt by now, `Accelerated DMA` is an eccentric lady. 
+She’s responsible for uploading and downloading image data directly from the GPU, avoiding the slow path of swizzling image data on the CPU and system memory.
+If you don’t tread carefully enough near her, you will get the wrong data, in the wrong place, at the wrong moment. 
+This will include a bunch of ara ara’s.
+Such was the fortune of `Sid Meier's Civilization VI`, a game that crashed on Vulkan, and rendered completely broken on OpenGL.
```

Comment:
```suggestion
As Maide has also learned by now, the `Accelerate DMA` fast path is fraught with peril.
`Accelerate DMA` is responsible for uploading and downloading image data directly from the GPU, avoiding the slow path of converting image data on the CPU and system memory.
But, it can only work when just the right conditions are met.
One game notably running into issues with the fast path was `Sid Meier's Civilization VI`, a game that crashed on Vulkan, and rendered completely broken on OpenGL.
```